src/gridcoin/staking/difficulty.cpp

```diff
@@ -370,6 +364,41 @@ double GRC::GetEstimatedTimetoStake(bool ignore_staking_status, double dDiff, do
         }
     }
 
+    // If dDiff = 0 from supplied argument (which is also the default), then derive a smoothed difficulty, otherwise
+    // let the supplied argument dDiff stand. The smoothed difficulty is derived via a two step process. There is a
+    // coupling between the desired block span to compute difficulty and essentially the stakeable balance: If the
+    // balance is low, the ETTS is expected to be relatively long, so it is appropriate to use a longer span to
+    // compute the difficulty. Conversely, if the stakeable balance is high, it is appropriate to use a corresponding
+    // short span so that the staking estimate reflects appropriate network conditions compared to the expected
+    // staking interval. Since this is a coupled problem, the approach here, which works reasonably well, uses the
+    // last hour (40 block) diff to bootstrap a span estimate from using the thumbrule estimate of ETTS with that diff,
+    // and then re-computes the diff using the block span, with clamp of [40, BLOCKS_PER_DAY], since it is silly to
+    // allow difficulty to become more sensitive than one hour of change, and is of minimal value to long term
+    // projections to use more than a day's history of diff. (Difficulty patterns tend to repeat on a daily basis.
+    // Longer term historical variations of more than a day are due to extraneous variables of which history is of
+    // little predictive value.)
+    if (!dDiff)
+    {
+        // First estimate the difficulty based on the last 40 blocks.
+        dDiff = GetAverageDifficulty(40);
+
+        // Compute an appropriate block span for the second iteration of dificulty computation based on the
+        // above diff calc. Clamp to no less than 40 (~1 hour) and no more than 960 (~1 day).
+        unsigned int nEstAppropriateDiffSpan = std::min((unsigned int) 960,
+                                                        std::max((unsigned int) 40,
+                                                                 (unsigned int) (9545.0 * BLOCKS_PER_DAY * COIN / nStakeableBalance * dDiff)));
```

Comment:
Where does the magic `9545.0` value come from?

Reply:
Ah. It is the actual thumbrule constand, which is usually rounded to 10000, although I think I picked up the wrong value. Going back to the bluepaper, it should be 9942.2056. Given this is a bootstrap swag anyway, we might as well round it off to 10000, which is the published thumbrule. I'll add a comment to the code on this to make it clear.